### PR TITLE
Mark dowantarray() as deprecated

### PR DESCRIPTION
**Pull request by ilmari**
This will make uses of the also-deprecated `GIMME` macro emit a compile-time warning.

Note that there are 3 modules still using `GIMME` in core, so this should not be merged until these are merged:

 * #20138 
 * pmqs/Compress-Raw-Zlib#19
 * pmqs/Compress-Raw-Bzip2#8

### CODE REVIEW

**Comment by jkeenan** [comment]
> This will make uses of the also-deprecated `GIMME` macro emit a compile-time warning.
> 
> Note that there are 3 modules still using `GIMME` in core, so this should not be merged until these are merged:
> 
>     * [Time-HiRes: use GIMME_V instead of the deprecated GIMME #20138](https://github.com/Perl/perl5/pull/20138)
> 
>     * [Use GIMME_V instead of the deprecated GIMME pmqs/Compress-Raw-Zlib#19](https://github.com/pmqs/Compress-Raw-Zlib/pull/19)
> 
>     * [Use GIMME_V instead of the deprecated GIMME pmqs/Compress-Raw-Bzip2#8](https://github.com/pmqs/Compress-Raw-Bzip2/pull/8)

Looks good to me, **but** ... we'll need new CPAN releases of Compress-Raw-Bzip2 and -Zlib which we can then synch into blead.

**Comment by jkeenan** [comment]
@ilmari, this pull request still needs review.  Could you resolve the merge conflicts that have developed so that we can move the discussion forward.

Thank you very much.

**Comment by demerphq** [comment]
Conflicts resolved.

**Review by demerphq — APPROVED** [review]

**Comment by demerphq** [comment]
@ilmari: Note all the patch links @jkeenan posted are applied. We should merge this once it passes CI.

**Comment by ilmari** [comment]
@demerphq The `Compress-Raw-{Zlib,Bzip2}` changes need to be released to CPAN and imported to blead before this can be merged.

**Comment by demerphq** [comment]
@ilmari acknowledged. Have you poked the owner to roll a release?

**Comment by demerphq** [comment]
@ilmari the modules have been released to CPAN by @pmqs.  See https://github.com/pmqs/Compress-Raw-Zlib/pull/19 for conversation on this.

**Pull request merged**
